### DISCUSSION TITLE
Best way to add blurhash to images?

### DISCUSSION THREAD

**Discussion by stelcodes** [pocketbase/pocketbase#5686] [Q&A] [answered]
I'd like to add a [blurhash](https://github.com/woltapp/blurhash) to uploaded images so I can show a blurred placeholder image while the image is loading. Usually these hashes are generated upon image upload and the hash is kept in the DB and coupled with the image URL. Has anyone integrated Blurhash into Pocketbase?

**Comment by KelvinConnor** [comment]
FWIW - I create the Blurhash on the client before the upload as I wanted to save processing resources on the server. I don't know how much an issue it would really be, but it might be something to consider if your solution is going to be processing lots of images. I also resize and compress the original on the client so it wasn't a big deal to add the hash creation.  The one thing I did discover I needed was to also save an aspect ratio along with the blurhash, else I could not size the placeholder correctly in my situation.

**Reply by stelcodes** [reply]
When you say on the client, do you mean you're uploading images to pocketbase via the SDK and not directly through the CMS interface? I was thinking about the latter in my case.

**Reply by KelvinConnor** [reply, marked as answer]
Correct. In my use case, a user uploads an image via my app. Within the app I use a library to generate the Blurhash (blurhash-dart), then I add it to the Pocketbase record in a field called 'blurhash.' My app can then use this to create a placeholder image while the app downloads the image over the net in the future. Both the creation of the hash and the utilization of the hash is all application code.

I don't know what you mean by the CMS specifically, but I believe you could do it on the Pocketbase side with JS or Go hooks (https://pocketbase.io/docs/js-event-hooks/#onrecordaftercreaterequest as example). The images are stored in folders next to the DB and you can get the name of the image file in the hook event. You would need to get the name of folders - top one is the Collection.id followed by the Record.id. Then run a process to generate the Hash and add it to the record. Again there may be performance issues with this approach. 

Another alternative would be just to run a scheduled background process on the server to look for images without a hash and then create it. The images simply would not have a blurhash until that process was run.

**Reply by stelcodes** [reply]
Ah ok that makes sense. I'll try to write a Go hook using one of the Go blurhash implementations. Thanks!
